components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/TypeConversionHandler.scala

```diff
@@ -47,8 +71,9 @@ object TypeConversionHandler {
     }
   }
 
-  private def handleStringToValueClassConversions(givenClass: TypedClass, superclassCandidate: TypedClass): Boolean = {
-    givenClass == Typed[String] && ValueClassesThatBeConvertedFromString.exists(ClassUtils.isAssignable(superclassCandidate.klass, _, true))
+  private def handleStringToValueClassConversions(givenType: SingleTypingResult, superclassCandidate: TypedClass): Boolean = {
+    givenType.isInstanceOf[TypedObjectWithValue] &&
+      valueClassesThatBeConvertedFromString(givenType.asInstanceOf[TypedObjectWithValue]).exists(ClassUtils.isAssignable(superclassCandidate.klass, _, true))
```

Comment:
Why instead of using filter andThen exists, use just exist? Thanks to that it will be simpler and you don't traverse the whole list each time